main.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"runtime"
+
+	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
+	// to ensure that exec-entrypoint and run can make use of them.
+	"github.com/operator-framework/operator-lib/leader"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorconfig "github.com/openshift/configure-alertmanager-operator/config"
+	operatormetrics "github.com/openshift/configure-alertmanager-operator/pkg/metrics"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	managedopenshiftiov1alpha1 "github.com/openshift/configure-alertmanager-operator/api/v1alpha1"
+	"github.com/openshift/configure-alertmanager-operator/controllers"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	scheme            = k8sruntime.NewScheme()
+	setupLog          = ctrl.Log.WithName("setup")
+	metricsHost       = "0.0.0.0"
+	metricsPort int32 = 8383
+)
+
+var log = logf.Log.WithName("cmd")
+
+func printVersion() {
+	log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+	log.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+}
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(configv1.AddToScheme(scheme))
```

Comment:
Better to use `configv1.Install(scheme)` as the `AddToScheme` is deprecated.

Reply:
Sounds good. I have made this change too.